src/components/CreateRecommendationModal/index.tsx

```diff
@@ -0,0 +1,318 @@
+        }
+    };
+
+    return (
+        <Modal
+            title="Create Recommendation"
+            open={isOpen}
+            onCancel={() => {
+                reset();
+                setImageUrl(''); // Clear image preview
+                setSelectedCountry('');
+                setSelectedLeague(undefined);
+                onClose();
+            }}
+            footer={null}
+            className={styles.modal}
+        >
+            <Form layout="vertical" className={styles.form} onFinish={handleSubmit(onSubmit)}>
+                <Form.Item
+                    label="Country"
+                    validateStatus={errors.country ? 'error' : ''}
+                    help={errors.country?.message}
+                >
+                    <AutoComplete
+                        options={countries.map((c) => ({ value: c.name }))}
+                        onSelect={(value) => {
+                            setSelectedCountry(value);
+                            setValue('country', value);
+                            setValue('league', '');
+                            setValue('stadium', '');
+                            setValue('homeTeam', '');
+                            setValue('awayTeam', '');
+                        }}
+                        placeholder="Select a country"
+                    />
+                </Form.Item>
+
+                <Form.Item label="League" validateStatus={errors.league ? 'error' : ''} help={errors.league?.message}>
+                    <AutoComplete
+                        options={leagues.map((l) => ({ value: l.league.name, id: l.league.id }))}
+                        onSelect={(value, option) => {
+                            setSelectedLeague(option.id);
+                            setValue('league', value);
+                            setValue('homeTeam', '');
+                            setValue('awayTeam', '');
+                        }}
+                        placeholder="Select a league"
+                        disabled={!selectedCountry}
+                    />
+                </Form.Item>
+
+                <Form.Item
+                    label="Stadium"
+                    validateStatus={errors.stadium ? 'error' : ''}
+                    help={errors.stadium?.message}
+                >
+                    <AutoComplete
+                        options={stadiums.map((venue) => ({ value: venue.name }))}
+                        onSelect={(value) => setValue('stadium', value)}
+                        placeholder="Select a stadium"
+                        disabled={!selectedCountry}
+                    />
+                </Form.Item>
+
+                <Form.Item
+                    label="Home Team"
+                    validateStatus={errors.homeTeam ? 'error' : ''}
+                    help={errors.homeTeam?.message}
+                >
+                    <AutoComplete
+                        options={teams.map((t) => ({ value: t.team.name }))}
```

Comment:
```suggestion
                        options={teams.map((option) => ({ value: option.team.name }))}
```
use meaningful var names